src/main/kotlin/org/rust/cargo/runconfig/wasmpack/WasmPackBuildTaskProvider.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.runconfig.wasmpack
+
+import com.intellij.execution.BeforeRunTask
+import com.intellij.execution.BeforeRunTaskProvider
+import com.intellij.execution.configurations.RunConfiguration
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.util.Key
+import com.intellij.task.ProjectTaskManager
+import com.intellij.util.execution.ParametersListUtil
+import org.rust.cargo.runconfig.buildtool.CargoBuildConfiguration
+import org.rust.cargo.runconfig.buildtool.CargoBuildManager
+import org.rust.cargo.runconfig.command.CargoCommandConfiguration
+import org.rust.cargo.runconfig.command.CargoCommandConfigurationType
+import org.rust.cargo.util.splitOnDoubleDash
+import org.rust.openapiext.project
+import java.util.concurrent.CompletableFuture
+import javax.swing.Icon
+
+class WasmPackBuildTaskProvider : BeforeRunTaskProvider<WasmPackBuildTaskProvider.BuildTask>() {
+    override fun getId(): Key<BuildTask> = ID
+    override fun getName(): String = "Cargo Build"
+    override fun getIcon(): Icon = AllIcons.Actions.Compile
+    override fun isSingleton(): Boolean = true
+
+    override fun createTask(runConfiguration: RunConfiguration): BuildTask? =
+        if (runConfiguration is WasmPackCommandConfiguration) BuildTask() else null
+
+    override fun executeTask(context: DataContext, configuration: RunConfiguration, environment: ExecutionEnvironment, task: BuildTask): Boolean {
+        if (configuration !is WasmPackCommandConfiguration) return false
+
+        val project = context.project ?: return false
+        val configurationArgs = ParametersListUtil.parse(configuration.command)
+        val (preArgs, postArgs) = splitOnDoubleDash(configurationArgs)
+        val configurationCommand = configurationArgs.firstOrNull() ?: return false
+
+        val buildCommand = ParametersListUtil.join(mutableListOf("build").apply {
+            addAll(postArgs)
+
+            if (configurationCommand == "test") {
+                addIfNotPresent("--tests")
```

Comment:
Maybe replace `linkedSetOf` with `mutableListOf` (and add `.toList()` at the end) to use `add` instead?